addons/sale/models/payment_transaction.py

```diff
@@ -49,54 +49,45 @@ def _set_pending(self, state_message=None, **kwargs):
             if tx.provider_id.code == 'custom':
                 for so in tx.sale_order_ids:
                     so.reference = tx._compute_sale_order_reference(so)
-            # send order confirmation mail.
-            sales_orders._send_order_confirmation_mail()
+            sales_orders._send_order_payment_executed_mail()
 
         return txs_to_process
 
-    def _check_amount_and_confirm_order(self):
-        """ Confirm the sales order based on the amount of a transaction.
+    def _post_process_payment_and_update_sale_order(self):
+        """ Update the sale order based on the amount of the transaction.
```

Comment:
The payment was already post_process at this point, you only check the amount of the transaction and confirm the order when necessary here =] The rename isn't necessary IMO